pkg/api/handlers/compat/containers_stats.go

```diff
@@ -22,14 +22,19 @@ func StatsContainer(w http.ResponseWriter, r *http.Request) {
 	decoder := r.Context().Value("decoder").(*schema.Decoder)
 
 	query := struct {
-		Stream bool `schema:"stream"`
+		Stream  bool `schema:"stream"`
+		oneShot bool `schema:"one-shot"` //added schema for one shot
 	}{
 		Stream: true,
 	}
 	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
 		utils.Error(w, "Something went wrong.", http.StatusBadRequest, errors.Wrapf(err, "failed to parse parameters for %s", r.URL.String()))
 		return
 	}
+	if query.Stream && query.oneShot { // mismatch. one-shot can only be passed with no-stream=true
```

Comment:
```suggestion
	if query.Stream && query.oneShot { // mismatch. one-shot can only be passed with stream=false
```
`no-stream` is not a valid query parameter.